src/js/jsx/sections/libraries/assets/Graphic.jsx

```diff
@@ -54,17 +57,51 @@ define(function (require, exports, module) {
             });
         },
 
+        /**
+         * Handle add layer from graphic asset
+         * @private
+         */
         _handleAdd: function () {
-            this.getFlux().actions.libraries.createLayerFromElement(this.props.element);
+            // TODO: new graphic is add to a fixed location. Ideally it should place the new graphic layer 
+            // at the center of the view port.
+            this.getFlux().actions.libraries.createLayerFromElement(this.props.element, { x: 100, y: 100 });
+        },
+        
+        /**
+         * Create preview of the dragged graphic asset under the cursor.
+         * @private
+         * @return {ReactComponent?}
+         */
+        _renderDragPreview: function () {
+            if (!this.props.dragPosition) {
```

Comment:
I found out that PS will halt the JavaScript before the end of the transform mode, so I set a 50ms delay to allow the draganddrop store finish its updating before placing the new layer. Please let me know if there is a better way to deal with this. https://github.com/adobe-photoshop/spaces-design/pull/1855/files#diff-768f12e2b3c6382be0ac86fe532542feR334